sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -278,13 +278,49 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def ansiDateTimeError(e: Exception): SparkDateTimeException = {
+    def extractDateTimeErrorInfo(e: Exception): (String, String, String) = {
+      val errorMessage = e.getMessage
+
+      val valuePattern = "Invalid value for ([A-Za-z]+) \\(valid values (.+)\\): (.+)".r
+      val datePattern = "Invalid date '[A-Z]+ ([0-9]+)'".r
+
+      errorMessage match {
+        case valuePattern(field, range, badValue) =>
+          val unit = field match {
+            case "Year" => "YEAR"
+            case "MonthOfYear" => "MONTH"
+            case "DayOfMonth" => "DAY"
+            case "HourOfDay" => "HOUR"
+            case "MinuteOfHour" => "MINUTE"
+            case "SecondOfMinute" => "SECOND"
+          }
+          val formattedRange = range.replace(" - ", " ... ")
+          (unit, formattedRange, badValue)
+        case datePattern(badDate) =>
+          ("DAY", "1 ... 28/31", badDate)
+        case _ =>
+          throw new SparkDateTimeException(
```

Comment:
@srielau @gengliangwang let's link issues to the proper tickets and not open duplicate tickets. The reason for not having a error condition is that `_LEGACY_ERROR_TEMP_####` errors do not show error condition. There is an existing ticket https://issues.apache.org/jira/browse/SPARK-49768 which was made to correct this. Also, could you please use the epic `ANSI by default` in JIRA if the change is related to ANSI, as this will make us not create multiple PRs/tickets for same issues. Also, @itholic leaving default branch like this does not solve the issue. We need to make sure we assign a proper error condition for all cases, which in this case we are not doing. (Example is if you try february 29th on a non-leap year) Ultimate goal is to remove `_LEGACY_ERROR_TEMP_2000`. @MaxGekk what do you think? I find it weird to have a method in QueryExecutionErrors throw completely new error just because of the error message parsing.